connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -453,11 +453,17 @@ public void putSessionKey(SessionKey sessionKey) {
         consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
 
         Map<String, Object> adminProps = new HashMap<>(originals);
-        NewTopic topicDescription = TopicAdmin.defineTopic(topic).
-                compacted().
-                partitions(1).
-                replicationFactor(config.getShort(DistributedConfig.CONFIG_STORAGE_REPLICATION_FACTOR_CONFIG)).
-                build();
+
+        Map<String, Object> topicSettings = null;
+        if (config instanceof DistributedConfig) {
+            topicSettings = ((DistributedConfig) config).configStorageTopicSettings();
+        }
```

Comment:
```suggestion
        Map<String, Object> topicSettings = config instanceof DistributedConfig
                ? ((DistributedConfig) config).configStorageTopicSettings()
                : Collections.emptyMap();
```
I know that `TopicAdmin#defineTopic` checks for `null`, but I think using `null` with collections is better to do when such optimization matters. Wdyt? 
(btw you don't have to use the ternary operator, I just added it to make the suggestion clear). 

Also, if you change here, please change in the other files too.